src/openms/source/CHEMISTRY/ModifiedNASequenceGenerator.cpp

```diff
@@ -90,209 +87,133 @@ namespace OpenMS
     }
   }
 
-  // static
-  void ModifiedNASequenceGenerator::applyVariableModifications(
-    const set<ConstRibonucleotidePtr>& var_mods,
-    const NASequence& seq,
-    size_t max_variable_mods_per_seq,
-    vector<NASequence>& all_modified_seqs,
-    bool keep_unmodified)
+
+  void ModifiedNASequenceGenerator::addModToSequences_(
+    vector<ModSeqInfo>& temp_seqs, Size n_temp_seqs,
+    vector<NASequence>& finished_seqs,
+    const function<bool(NASequence&, Int&)>& applyMod)
   {
-    // no variable modifications specified or no variable mods allowed? no compatibility map needs to be build
-    if (var_mods.empty() || max_variable_mods_per_seq == 0)
+    for (Size i = 0; i < n_temp_seqs; ++i)
```

Comment:
can you extend the documentation of this function?

Reply:
I've added a comment. There's already documentation in the header file.